lib/mutant/ast/meta/send.rb

```diff
@@ -59,6 +66,25 @@ def receiver_possible_top_level_const?
           Const.new(receiver).possible_top_level?
         end
 
+      private
+
+        # Test if node is `proc { ... }`
+        #
+        # @return [Boolean]
+        def naked_proc?
+          !receiver && selector.equal?(:proc)
+        end
+
+        # Test if node is `Proc.new { ... }`
+        #
+        # @return [Boolean]
+        def proc_new?
+          receiver                &&
+            selector.equal?(:new) &&
+            n_const?(receiver)    &&
+            Const.new(receiver).name.equal?(:Proc)
```

Comment:
I'd love if we could at some point simply inject a value probe to evaluate the constant in the environment to simply do `#equal?(::Proc)`.